src/Microsoft.AspNetCore.Mvc.TagHelpers/ScriptTagHelper.cs

```diff
@@ -316,80 +318,90 @@ private void BuildFallbackBlock(TagHelperAttributeList attributes, TagHelperCont
                         var attribute = attributes[i];
                         if (!attribute.Name.Equals(SrcAttributeName, StringComparison.OrdinalIgnoreCase))
                         {
-                            var encodedKey = JavaScriptEncoder.Encode(attribute.Name);
-                            var attributeValue = GetAttributeValue(attribute.Value);
-                            var encodedValue = JavaScriptEncoder.Encode(attributeValue);
-
-                            AppendAttribute(builder, encodedKey, encodedValue, escapeQuotes: true);
+                            //var attributeValue = GetAttributeValueAsContent(attribute.Value);
+                            WriteAttribute(attribute.Name, attribute.Value, attribute.Structure, StringWriter);
                         }
                         else
                         {
                             addSrc = false;
-                            AppendEncodedVersionedSrc(attribute.Name, src, builder, generateForDocumentWrite: true);
+                            WriteEncodedVersionedSrc(attribute.Name, src, attribute.Structure, StringWriter);
```

Comment:
As done elsewhere in this `class`, create a new `TagHelperAttribute` with the new value and just tell that to write itself.